moveit_core/robot_state/test/robot_state_test.cpp

```diff
@@ -567,72 +571,149 @@ TEST_F(OneRobot, FK)
+  robot_state->setToDefaultValues();
+  ja = robot_state->getVariablePosition("joint_a");
+  robot_state->setVariablePosition("joint_a", ja + 1.0);
+  jc = robot_state->getVariablePosition("joint_c");
+  robot_state->setVariablePosition("joint_c", jc - 0.01);
   traj.push_back(robot_state);
+
+  // 6th waypoint with a large jump of 1.01 in first prismatic joint
+  robot_state.reset(new robot_state::RobotState(*robot_state));
+  robot_state->setToDefaultValues();
+  ja = robot_state->getVariablePosition("joint_a");
+  robot_state->setVariablePosition("joint_a", ja + 1.0);
+  jc = robot_state->getVariablePosition("joint_c");
+  robot_state->setVariablePosition("joint_c", jc + 1.0);
+  traj.push_back(robot_state);
+
+  // 7th waypoint with no jump
+  robot_state.reset(new robot_state::RobotState(*robot_state));
+  robot_state->setToDefaultValues();
+  ja = robot_state->getVariablePosition("joint_a");
+  robot_state->setVariablePosition("joint_a", ja + 1.0);
+  jc = robot_state->getVariablePosition("joint_c");
+  robot_state->setVariablePosition("joint_c", jc + 1.0);
+  traj.push_back(robot_state);
+
+  return traj.size();
 }
 
 TEST_F(OneRobot, testAbsoluteJointSpaceJump)
 {
-  const robot_model::JointModelGroup* joint_model_group = robot_model->getJointModelGroup("base_from_base_to_tip");
+  const robot_model::JointModelGroup* joint_model_group = robot_model->getJointModelGroup("base_from_base_to_e");
   std::vector<std::shared_ptr<robot_state::RobotState>> traj;
-
-  // direct call of absolute version
-  generateTestTraj(traj, robot_model, joint_model_group);
-  double fraction = robot_state::RobotState::testAbsoluteJointSpaceJump(joint_model_group, traj, 1.0, 1.0);
-  EXPECT_EQ(traj.size(), 3);  // traj should be cut
-  EXPECT_NEAR(fraction, 3. / 4., 0.01);
-
-  // indirect call
-  generateTestTraj(traj, robot_model, joint_model_group);
+  // generate a test trajectory of len 4 with all zeros except one large jump at the last waypoint
+  std::size_t traj_len = generateTestTraj(traj, robot_model, joint_model_group);
+
+  // The absolute jump of -1.01 occurs at the 4th waypoint for a revolute joint and at the 5th waypoint for in a
+  // prismatic joint
+  std::size_t expected_absolute_revolute_jump_traj_len = 4;
+  std::size_t expected_absolute_prismatic_jump_traj_len = 5;
+  std::size_t expected_full_traj_len = 7;
+
+  // Containers for results
+  std::size_t full_traj_len;
+  double fraction;
+
+  // Direct call of absolute version
+  full_traj_len = generateTestTraj(traj, robot_model, joint_model_group);
+  fraction = robot_state::RobotState::testAbsoluteJointSpaceJump(joint_model_group, traj, 1.0, 1.0);
+  EXPECT_EQ(full_traj_len, expected_full_traj_len);                  // full traj should be 7 waypoints long
+  EXPECT_EQ(traj.size(), expected_absolute_revolute_jump_traj_len);  // traj should be cut
+  EXPECT_NEAR(fraction, (double)expected_absolute_revolute_jump_traj_len / (double)full_traj_len, 0.01);
```

Comment:
Do you think, this is more readable than `EXPECT_NEAR(fraction, 4. / 7., 0.01);`?

Reply:
i really prefer naming constants in variables rather than magic numbers inline with a test

Reply:
I'm with Dave on this one.